src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+ */
+void EnSlime_CheckACHit(EnSlime* this, PlayState* play) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F684.s")
+        if ((this->damageEffectType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) ||
+            !(this->collider.info.acHitInfo->toucher.dmgFlags & 0xDB0B3)) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F6CC.s")
+            EnSlime_UpdateIceEffects(this, play);
+            if ((this->actor.params == EN_SLIME_TYPE_BLUE) &&
+                (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_ICEARROW)) {
+                // Blue chuchus take no damage from ice arrows.
+                this->actor.colChkInfo.damage = 0;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F8B4.s")
+            if (Actor_ApplyDamage(&this->actor) == 0) {
+                Actor_SetDropFlag(&this->actor, &this->collider.info);
+                Enemy_StartFinishingBlow(play, &this->actor);
+                this->actor.flags &= ~ACTOR_FLAG_1;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F8E0.s")
+            if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_BLUNT) {
+                EnSlime_ReactToBluntHit(this);
+            } else if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_STUN) {
+                this->timer = 40;
+                Actor_SetColorFilter(&this->actor, 0, 255, 0x2000, 40);
+                Actor_PlaySfx(&this->actor, NA_SE_EN_COMMON_FREEZE);
+                EnSlime_EnterStunnedState(this);
+            } else if (this->actor.colChkInfo.damageEffect != ENSLIME_DMGTYPE_HOOKSHOT) {
+                if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_ICEARROW) {
+                    if (this->actor.params == EN_SLIME_TYPE_BLUE) {
+                        EnSlime_InitIceBlock(this);
+                    } else {
+                        EnSlime_InitIceStun(this);
+                        if (this->actor.colChkInfo.health == 0) {
+                            this->timer = 3;
+                            this->collider.base.acFlags &= ~AC_ON;
+                        }
+                        EnSlime_EnterStunnedState(this);
+                    }
+                } else {
+                    if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_LIGHT) {
+                        this->effectAlpha = 4.0f;
+                        this->effectScale = 0.4f;
+                        this->damageEffectType = ACTOR_DRAW_DMGEFF_LIGHT_ORBS;
+                        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->collider.info.bumper.hitPos.x,
+                                    this->collider.info.bumper.hitPos.y, this->collider.info.bumper.hitPos.z, 0, 0, 0,
+                                    4);
```

Comment:
There are defines for the parameters when spawning `ACTOR_EN_CLEAR_TAG`:
```suggestion
                        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->collider.info.bumper.hitPos.x,
                                    this->collider.info.bumper.hitPos.y, this->collider.info.bumper.hitPos.z, 0, 0, 0,
                                    CLEAR_TAG_LARGE_LIGHT_RAYS);
```